python/sdist/amici/sbml_import.py

```diff
@@ -677,8 +714,19 @@ def _gather_base_locals(self):
         ):
             if not c.isSetId():
                 continue
-
-            self.add_local_symbol(c.getId(), _get_identifier_symbol(c))
+            if c.getId() in hardcode_symbols:
+                if self.sbml.getRuleByVariable(c.getId()) is not None:
+                    raise ValueError(
+                        f"Cannot hardcode symbol `{c.getId()}` that is a rule target."
+                    )
+                if self.sbml.getInitialAssignment(c.getId()):
```

Comment:
add event assignements. What about assignment rules?

Reply:
Event assignments were missing, true. Assignment rules were handled by `getRuleByVariable`.  Now changed to checking for the `constant` attribute which covers both.